x-pack/plugins/alerting/server/rules_client/methods/bulk_edit.ts

```diff
@@ -302,209 +306,419 @@ async function bulkEditOcc<Params extends RuleTypeParams>(
+      await getUpdatedAttributesFromOperations(context, operations, rule, ruleType);
+
+    validateScheduleInterval(
+      context,
+      attributes.schedule.interval as string,
+      ruleType.id,
+      attributes.id as string
+    );
+
+    const { modifiedParams: ruleParams, isParamsUpdateSkipped } = paramsModifier
+      ? await paramsModifier(attributes.params as Params)
+      : {
+          modifiedParams: attributes.params as Params,
+          isParamsUpdateSkipped: true,
+        };
+
+    // If neither attributes nor parameters were updated, mark
+    // the rule as skipped and continue to the next rule.
+    if (isAttributesUpdateSkipped && isParamsUpdateSkipped) {
+      skipped.push({
+        id: rule.id,
+        name: rule.attributes.name,
+        skip_reason: 'RULE_NOT_MODIFIED',
+      });
+      return;
+    }
+
+    // validate rule params
+    const validatedAlertTypeParams = validateRuleTypeParams(ruleParams, ruleType.validate?.params);
+    const validatedMutatedAlertTypeParams = validateMutatedRuleTypeParams(
+      validatedAlertTypeParams,
+      rule.attributes.params,
+      ruleType.validate?.params
+    );
 
-          // get notifyWhen
-          const notifyWhen = getRuleNotifyWhenType(
-            attributes.notifyWhen ?? null,
-            attributes.throttle ?? null
-          );
+    const {
+      actions: rawAlertActions,
+      references,
+      params: updatedParams,
+    } = await extractReferences(
+      context,
+      ruleType,
+      ruleActions.actions,
+      validatedMutatedAlertTypeParams
+    );
 
-          const updatedAttributes = updateMeta(context, {
+    const { apiKeyAttributes } = await prepareApiKeys(
+      context,
+      rule,
+      ruleType,
+      apiKeysMap,
+      attributes,
+      hasUpdateApiKeyOperation,
+      username
+    );
+
+    const { updatedAttributes } = updateAttributes(
+      context,
+      attributes,
+      apiKeyAttributes,
+      updatedParams,
+      rawAlertActions,
+      username
+    );
+
+    rules.push({
+      ...rule,
+      references,
+      attributes: updatedAttributes,
+    });
+  } catch (error) {
+    errors.push({
+      message: error.message,
+      rule: {
+        id: rule.id,
+        name: rule.attributes?.name,
+      },
+    });
+    context.auditLogger?.log(
+      ruleAuditEvent({
+        action: RuleAuditAction.BULK_EDIT,
+        error,
+      })
+    );
+  }
+}
+
+async function ensureAuthorizationForBulkUpdate(
+  context: RulesClientContext,
+  operations: BulkEditOperation[],
+  rule: SavedObjectsFindResult<RawRule>
+): Promise<void> {
+  for (const operation of operations) {
+    const { field } = operation;
+    if ((field === 'snoozeSchedule' || field === 'apiKey') && rule.attributes.actions.length > 0) {
+      try {
+        await context.actionsAuthorization.ensureAuthorized('execute');
+      } catch (error) {
+        throw Error(`Rule not authorized for bulk ${field} update - ${error.message}`);
+      }
```

Comment:
If we meet certain condition first time we will check actions authorization. As I understood it's enough to do once. So we can `break` the look in that case.